cluster-autoscaler/cloudprovider/clusterapi/clusterapi_utils.go

```diff
@@ -187,7 +194,14 @@ func parseCPUCapacity(annotations map[string]string) (resource.Quantity, error)
 func parseMemoryCapacity(annotations map[string]string) (resource.Quantity, error) {
 	// the value for the memoryKey is expected to have the unit type included,
 	// eg "1024Mi". if only a number is present, we add the suffix "Mi".
-	val, exists := annotations[memoryKey]
+	var val string
+	var exists bool
+	val, exists = annotations[memoryKey]
+	if !exists {
+		// TODO(alberto): drop this fallback after removing the deprecated key from aws/azure/gcp controllers
+		val, exists = annotations[memoryKeyDeprecated]
+	}
+
 	if exists && val != "" {
 		// TODO remove this check once we ensured that the providers are using the correct values
 		if _, err := strconv.Atoi(val); err == nil {
```

Comment:
Should this bit be inside the legacy part? It was my understanding that with this change we would be dropping the assumption that values were `Mi`?

Reply:
++, we should probably just change this to "M" if we are making the same assumption in the providers.

Reply:
>Should this bit be inside the legacy part? It was my understanding that with this change we would be dropping the assumption that values were Mi?

yes, left it to keep it simple, we can just drop it along with the deprecated key after this and the providers counter part prs merge